### PR TITLE
fix(DTFS2-7292): downgraded MSSQL to 2022

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
A recent update on Microsoft MSSQL version especially `2022` have path and driver updates.
This has broken the pipeline due to absence of any health check for `dtfs-sql` container pipeline debugging has been inaccurate.

This PR aims to mitigate both the issues.

## Resolution :heavy_check_mark:
* Downgraded MSSQL docker image to explicit version `mcr.microsoft.com/mssql/server:2022-CU13-ubuntu-22.04`
* Refactored `init.sh` for MSSQL server boot.
* Added health check to `dtfs-sql` container.

## Miscellaneous :heavy_plus_sign:
* Renamed `dtfs-submissions-data` to `dtfs-mongo` for consistent naming convention.

### CODE REVIEW

File: libs/common/src/sql-db-connection/init.sh
Comment (JoshBinns2000) [inline libs/common/src/sql-db-connection/init.sh:1]: I don't think we should change this file. What was the reason to remove the start-up time limit?

File: libs/common/src/sql-db-connection/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Base image
-FROM mcr.microsoft.com/mssql/server
+FROM mcr.microsoft.com/mssql/server:2019-latest
```
Comment (JoshBinns2000) [inline libs/common/src/sql-db-connection/Dockerfile:2]: Is there any reason to not use the more recent `mcr.microsoft.com/mssql/server:2022-CU13-ubuntu-22.04` image?

File: libs/common/src/sql-db-connection/init.sh [hunk 2]
```diff
@@ -1,54 +1,42 @@
...
-do
-  current_time=$(get_now_unix_timestamp)
-  elapsed_seconds=$((current_time - start_time))
-
-  if [ $elapsed_seconds -gt $time_limit_seconds ]; then
-    echo "❌ Error: Failed to create database '$SQL_DB_NAME' - SQL Server did not start within $time_limit_seconds seconds."
-    exit 1
-  fi
+# Executes SQL query
+query() {
```
Comment (JoshBinns2000) [inline libs/common/src/sql-db-connection/init.sh:8]: If we want to change `command` to `query`, I think calling the function `execute_sql_query` is probably more clear

File: libs/common/src/sql-db-connection/init.sh [hunk 3]
```diff
@@ -1,54 +1,42 @@
...
-  echo "❌ MSSQL failed to start after $elapsed_seconds seconds. Waiting for $interval_seconds more seconds."
-  sleep $interval_seconds
-done
+# Initial DB setup
+initialise() {
+  # 1. Database
+  query "CREATE DATABASE [$SQL_DB_NAME];"
+  query "USE [$SQL_DB_NAME];"
 
-echo "✅ MSSQL initialised, creating database '$SQL_DB_NAME'..."
```
Comment (JoshBinns2000) [inline libs/common/src/sql-db-connection/init.sh:36]: I think this logging is quite helpful when looking inside the container as the other logs aren't very descriptive. Is there a reason to remove them?

File: libs/common/src/sql-db-connection/init.sh
Comment (abhi-markan) [reply libs/common/src/sql-db-connection/init.sh:1]: We discussed this during our conversation.

File: libs/common/src/sql-db-connection/init.sh (hunk 3 above)
Comment (abhi-markan) [reply libs/common/src/sql-db-connection/init.sh:36]: I have kept the logging in place and added an extra line to distinguish between when the MSSQL service is running and when it has been initialised.

**Review by JoshBinns2000 — APPROVED** [review]

**Pull request merged**